content/learning-paths/smartphones-and-mobile/android_webgpu_dawn/5-render-a-simple-3D-object-part-1.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Render a simple 3D object - part 1
+weight: 6
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+## Command Queue
+
+Graphic applications have to deal with 2 processors, one CPU and other GPU. These 2 processors run on different timelines. For optimal performance  commands intended for the GPU are batched and fired through a command queue. The GPU consumes this queue whenever it is ready, and this way processors minimize the time spent idling for their sibling to respond. WebGPU device has a single queue, which is used to send both commands and data. We can get it with `wgpuDeviceGetQueue`.
+WebGPU offers 3 different ways to submit work to this queue:
+
+* wgpuQueueSubmit
+* wgpuQueueWriteBuffer
+* wgpuQueueWriteTexture
+
+{{% notice Note %}}
+Other graphics API allow one to build multiple queues per device, and future version of WebGPU might as well. But for now, one queue is already more than enough for us to play with! If you wish to learn more, please refer to the [Command Queue](https://eliemichel.github.io/LearnWebGPU/getting-started/the-command-queue.html)
+{{% /notice %}}
+
+## Getting started to render a 3D object
+
+WebGPU is a very simple system. All it does is run 3 types of functions on the GPU: Vertex Shaders, Fragment Shaders, and Compute Shaders.
```

Comment:
This is really overselling the simplicity i.m.o. For one, it also provides fixed-functionality stages in the graphics pipeline and utility functions for copying data around.
I would call them "programs" rather than functions.